examples/i18n/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+// Some Clippy linter rules are ignored for the sake of simplicity.
+#![allow(clippy::needless_pass_by_value, clippy::trivially_copy_pass_by_ref)]
+
+use seed::{prelude::*, *};
+
+use fluent::{FluentArgs, FluentBundle, FluentResource, FluentValue};
+use unic_langid::LanguageIdentifier;
+
+use enum_iterator::IntoEnumIterator;
+use std::borrow::Borrow;
+
+mod resource;
+
+// ------ ------
+//     Init
+// ------ ------
+
+fn init(_: Url, _: &mut impl Orders<Msg>) -> Model {
+    let mut model = Model::default();
+    change_lang(&mut model);
+    model
+}
+
+// ------ ------
+//     Model
+// ------ ------
+#[allow(non_camel_case_types)]
+#[derive(Clone, IntoEnumIterator, PartialEq)]
+enum Lang {
```

Comment:
Please move `Lang` below `Model` and add comment `// ------ Lang ------` above it (you can see it in other examples).

You want to see the "parent" item first (`Model` in this case) to see usage (`lang: Lang`) and then you want to see the definition (`enum Lang...`). It's not super clear when there is only one "child", but you can't load all children into your short-term memory when there are many child items. It would cause increased coginitive load and jumping from one place in code to another and back.
The same rule is valid for all nested items like nested `view`s and `update` helpers.

Reply:
Good reasoning. Done.